src/types.ts

```diff
@@ -166,6 +166,7 @@ export type DataSheetGridProps<T> = {
   columns?: Partial<Column<T, any, any>>[]
   gutterColumn?: SimpleColumn<T, any> | false
   stickyRightColumn?: SimpleColumn<T, any>
+  rowKey?: string
```

Comment:
```suggestion
  rowKey?: string | (rowDate: T, index: number) => string
```

I feel like it should support a function for non-object rows.
Then we can have a default value of `(_, index) => String(index)` and avoid special treatment in the code